### ISSUE TITLE
Silent build failure

### ISSUE THREAD

**Issue by tony19** [Polymer/polymer-build#119]
### Description
The build silently fails for some reason. My [repro](https://github.com/tony19-sandbox/polymer-build-issue-88.git) intentionally has missing files in HTML imports to trigger #88, but that issue was fixed by #111 in `polymer-build` 0.7.0, so this bug might be a symptom of a different bug.

### Versions & Environment
- polymer-cli: 0.18.0-alpha.9
- polymer-build: 0.7.0
- node: 7.4.0
- Operating System: macOS Sierra 10.12

#### Steps to Reproduce

 1. Clone https://github.com/tony19-sandbox/polymer-build-issue-88.git
 2. In the newly created directory, run the following commands:

        npm install
        bower install
        polymer build

 3. Observe the build exits prematurely.

#### Expected Results
The build errors out with "file not found".

#### Actual Results
The build aborts with exit code 0:

```shell
$ polymer build 
warn:    "sourceGlobs" config option has been renamed to "sources" and will no longer be supported in future versions
warn:    "includeDependencies" config option has been renamed to "extraDependencies" and will no longer be supported in future versions
info:    Preparing build...
info:    Building application...
info:    Generating build/unbundled...
info:    Generating build/bundled...
$
```

```shell
$ gulp build
warn:    "sourceGlobs" config option has been renamed to "sources" and will no longer be supported in future versions
warn:    "includeDependencies" config option has been renamed to "extraDependencies" and will no longer be supported in future versions
[00:03:50] Using gulpfile ~/src/polymer-build-issue-88/gulpfile.js
[00:03:50] Starting 'build'...
Deleting build directory...
Analyzing build dependencies...
$
```

This also occurs in [Travis](https://travis-ci.org/tony19-sandbox/polymer-build-issue-88/jobs/197565652) for Node 4 through 7:
![screen shot 2017-02-02 at 2 11 49 am](https://cloud.githubusercontent.com/assets/26580/22541616/02a74e4e-e8ed-11e6-8759-eee786e4fe3a.png)

**Comment by tony19** [comment]
@FredKSchott I just noticed that my repro's `README` was missing a step to install Bower dependencies. Could that be why you weren't able to reproduce?

**Comment by web-padawan** [comment]
I can confirm this issue on Windows 7 with clean npm install.
Same happens with our project, which is a lot more bigger, but the output looks the same.
Especially this case is likely to happen for us too:

```
debug:   dependency is a source file, ignoring...
{ dep: 'src/ost-stars/ost-stars.html' }
```

This is exactly the case which I try to gebug since `0.6.0-alpha.1` with no success,
All works fine with 0.5.1, event though bundling continues more than 2 minutes.

Any ideas about kind of regression which could be introduced when switching to `analyzer first` approach?

**Comment by FredKSchott** [comment]
@web-padawan there is nothing wrong with that debug statement you're seeing, it's trying to tell you that the analyzer has found an import, but that it doesn't need to go out and double-load it since it's already loaded as a source file. That could definitely be rewritten to be more clear.

**Comment by FredKSchott** [comment]
@tony19 finally, I can reproduce! For some reason `neon-animation` wasn't being installed when I ran `bower install`.

Looking into this now...

**Comment by FredKSchott** [comment]
TIL a lot about streams, specifically the `highWaterMark` option. It turns out that our analyze stream will only buffer 16 files in the build stream while it analyzed, and would pause if it ever had more than that. The analyze stream was basically starving itself by pausing the entire source stream when it was too backed up waiting for a file to come through *that exact same source stream*. This also explains why it only appears on more complex projects (simple ones don't hit the 16-object limit).

The long-term solution is to move analysis out of the stream. Now that the `BuildAnalyzer` is no longer a stream itself there's no reason to block the sources/dependencies stream while we analyze. This should fix the bug and make builds more efficient.

The short-term solution can be to increase the `highWaterMark` so that it no longer blocks. I'm going to create a PR to do this today so that we can get a fix out sooner, and then anyone can work on the long-term solution when they have room on their plate.

@tony19 thanks again for your patience and help throughout this investigation, excited to get this fixed for you and everyone else!

**Comment by tony19** [comment]
@FredKSchott Awesome, thank *you*!

**Comment by FredKSchott** [comment]
Just released via `v0.7.1`

**Comment by tony19** [comment]
@FredKSchott Sorry to be a PITA, but it seems there are some regressions in the new release. While it no longer silently fails, it's also ignoring missing HTML imports (#124).
